### PR TITLE
enso-8194 Copy and pasting a single node

### LINKED ISSUE #8194: Copy and pasting a single node
See https://github.com/enso-org/enso/pull/7618 for reference.

- <kbd>cmd + c</kbd> and <kbd>cmd + v</kbd> (<kbd>control</kbd> on non-Mac systems) shortcuts allow copying and pasting a single selected node.
- Node metadata should also be copied so the state of visualizations is preserved (the position, of course, should not be preserved)
- [Clipboard pickling](https://github.com/w3c/editing/blob/gh-pages/docs/clipboard-pickling/explainer.md) can be used for our custom clipboard content format.
- Copying node should also copy its code as `text/plain` payload, so it should be possible to paste the copied code to the external program.
- The other way (pasting code from external programs) should not be supported for security reasons.

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description
Fixes  #8194

[Peek 2023-11-13 14-15.webm](https://github.com/enso-org/enso/assets/1428930/22882487-7288-4eb7-af08-6539060b4967)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by vitvakatu** [comment]
Please note that this implementation stores nodes in a custom JSON-based format with `text/plain` MIME-type, which means any external program will receive a bunch of unnecessary metadata when pasting.
The implementation in the old GUI was using two payloads: 
1. `text/plain` contains only the code. It is used when pasting in other programs.
2. `web application/enso` containing our custom format with metadata.

This was a feature specifically requested on the demo.

Also, we’re not handling node positions here – the node is always placed under the cursor, even if another node is already there. We don‘t have the necessary API for this issue yet, but it must be remembered.

File: app/gui2/src/components/GraphEditor.vue [hunk 1]
```diff
@@ -225,6 +233,44 @@ function getNodeContent(id: ExprId): string {
...
+}
+
+/// Copy the content of the selected node to the clipboard.
+function copyNodeContent() {
+  const id = nodeSelection.selected.values().next().value
+  const node = graphStore.nodes.get(id)
+  if (node == null) return
+  const content = node.rootSpan.repr()
+  const metadata = projectStore.module?.getNodeMetadata(id)
+  const copiedNode: CopiedNode = { expression: content, metadata }
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor.vue:309]: The stored format assumes a single node. It is fine to support copying only one for now, but let's make some room for easily extending it later, such as making the top-level object contain an array of nodes already (e.g. `{ nodes: CopiedNode[] }`).

**Review by Frizi — CHANGES_REQUESTED** [review]
Please address the mime type issues pointed out by @vitvakatu.

**Review by Frizi — APPROVED** [review]

**Comment by somebody1234** [comment]
@MichaelMauderer just a heads up, this is ready to merge (all required checks are passing too) but there are new merge conflicts

**Pull request merged**